### PR TITLE
Fastlane secrets plugin

### PR DESCRIPTION
**Pull request by amlcurran**
Adds the fastlane secrets plugin!

The readme contains information about how to run the plugin, although doesn't yet say _how_ to add the plugin to a project, that will come with some further refinements :) 

screen shot 2018-06-19 at 17 55 41

### CODE REVIEW

File: Fastlane Secrets/fastlane-plugin-generate_secrets/lib/fastlane/plugin/generate_secrets/actions/generate_secrets_action.rb [hunk 1]
```diff
@@ -1,11 +1,40 @@
...
+        [:key_prefix, :class_name].each do |required_parameter|
+          if params[required_parameter].nil?
+            UI.user_error!("#{required_parameter} is a required parameter for generate_secrets")
+          end
+        end
+
+        file_name = params[:file_name].nil? ? params[:class_name] : params[:file_name]
+        access_modifier = params[:public] ? "public " : ""
+        class_name = params[:class_name]
+        declaration = params[:use_extension] ? "extension" : "class"
```
Comment (amlcurran) [inline Fastlane Secrets/fastlane-plugin-generate_secrets/lib/fastlane/plugin/generate_secrets/actions/generate_secrets_action.rb]: these lines do the configuration

File: Fastlane Secrets/fastlane-plugin-generate_secrets/lib/fastlane/plugin/generate_secrets/actions/generate_secrets_action.rb [hunk 2]
```diff
@@ -1,11 +1,40 @@
...
+
+        path = params[:path]
+        directory = path ? File.expand_path(path, Dir.pwd) : Dir.pwd
+        final_path = File.join(directory, "#{file_name}.swift")
+        FileUtils.mkdir_p(directory)
```
Comment (amlcurran) [inline Fastlane Secrets/fastlane-plugin-generate_secrets/lib/fastlane/plugin/generate_secrets/actions/generate_secrets_action.rb]: these lines make sure we have a directory to write to

File: Fastlane Secrets/fastlane-plugin-generate_secrets/lib/fastlane/plugin/generate_secrets/actions/generate_secrets_action.rb [hunk 3]
```diff
@@ -1,11 +1,40 @@
...
+          file.write("#{access_modifier}#{declaration} #{class_name} {\n")
+          file.write("\n")
+          ENV.each do |key, value|
+            next unless key.to_s.start_with?(params[:key_prefix])
+            extracted_key = key.sub(params[:key_prefix], "")
+            file.write("  #{access_modifier}static let #{extracted_key} = \"#{value}\"")
+            file.write("\n")
+          end
+          file.write("}\n")
+        end
```
Comment (amlcurran) [inline Fastlane Secrets/fastlane-plugin-generate_secrets/lib/fastlane/plugin/generate_secrets/actions/generate_secrets_action.rb]: this writes the file!

File: Fastlane Secrets/Fastlane Secrets/AppDelegate.swift [hunk 4]
```diff
@@ -22,8 +22,9 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         label.trailingAnchor.constraint(equalTo: viewController.view.safeAreaLayoutGuide.trailingAnchor).isActive = true
         label.topAnchor.constraint(equalTo: viewController.view.safeAreaLayoutGuide.topAnchor).isActive = true
 
-        label.text = "Secret key will go here!"
+        label.text = Secrets.apiKey
```
Comment (amlcurran) [inline Fastlane Secrets/Fastlane Secrets/AppDelegate.swift]: Note the secrets file is not checked in to the repo, but could be generated using fastlane on a CI, for example

**Comment by florianmski** [comment]
Waddup with that PR? Should we merge? close? cc @amlcurran @gbasile

**Pull request merged**